src/werkzeug/datastructures.py

```diff
@@ -1755,6 +1768,46 @@ def to_header(self):
             result.append(value)
         return ",".join(result)
 
+    @t.overload
+    @classmethod
+    def parse_header(cls, value: t.Optional[str]) -> "Accept":
+        ...
+
+    @t.overload
+    @classmethod
+    def parse_header(cls, value: t.Optional[str]) -> _TAnyAccept:
+        ...
+
+    @classmethod
+    def parse_header(cls, value: t.Optional[str]) -> _TAnyAccept:
+        """Parses an HTTP Accept-* header.  This does not implement a complete
+        valid algorithm but one that supports at least value and quality
+        extraction.
+
+        Returns a new :class:`Accept` object (basically a list of ``(value, quality)``
+        tuples sorted by the quality with some additional accessor methods).
+
+        The second parameter can be a subclass of :class:`Accept` that is created
+        with the parsed values and returned.
+
+        :param value: the accept header string to be parsed.
+        :param cls: the wrapper class for the return value (can be
```

Comment:
I don't think it is typical to annotate the `cls` argument for a classmethod. Also I think it would be difficult to pass a subclass for the `cls` value.